### PR TITLE
pkp/pkp-lib#2571 consider abstract word count limit 

### PR DESCRIPTION
**Pull request by bozana**
s. https://github.com/pkp/pkp-lib/issues/2571

### CODE REVIEW

File: classes/submission/SubmissionMetadataFormImplementation.inc.php [hunk 1]
```diff
@@ -33,6 +33,33 @@ function _getAbstractsRequired($submission) {
...
+	 *
+	 * @copydoc PKPSubmissionMetadataFormImplementation::addChecks()
+	 */
+	function addChecks($submission) {
+		parent::addChecks($submission);
+		$sectionDao = DAORegistry::getDAO('SectionDAO');
+		$section = $sectionDao->getById($submission->getSectionId());
+		$wordNo = $section->getAbstractWordCount();
+		if (isset($wordNo) && $wordNo > 0) {
+			$this->_parentForm->addCheck(new FormValidatorCustom($this->_parentForm, 'abstract', 'required', 'submission.submit.form.wordCountAlert', create_function('$abstract, $wordNo', 'foreach ($abstract as $localizedAbstract) {return count(explode(" ",trim(strip_tags($localizedAbstract)))) <= $wordNo; }'), array($wordNo)));
```
Comment (NateWr) [inline classes/submission/SubmissionMetadataFormImplementation.inc.php]: Same `wordNo` to `wordCount` here?

File: locale/en_US/submission.xml [hunk 2]
```diff
@@ -22,6 +22,7 @@
 	<message key="submission.submit.newSubmissionMultiple">Start a New Submission in</message>
 	<message key="submission.submit.newSubmissionSingle">New Submission</message>
 	<message key="grid.action.workflow">Submission Workflow</message>
+	<message key="submission.abstract.wordNo.description">The abstract can contain maximal {$wordNo} words.</message>
```
Comment (NateWr) [inline locale/en_US/submission.xml]: > The abstract must be {$wordCount} words or less.

File: locale/en_US/submission.xml [hunk 3]
```diff
@@ -70,6 +71,7 @@
 	<message key="submission.submit.form.authorRequiredFields">The first name, last name, and email address of each author is required.</message>
 	<message key="submission.submit.form.titleRequired">Please enter the title of your article.</message>
 	<message key="submission.submit.form.abstractRequired">Please enter your article's abstract.</message>
+	<message key="submission.submit.form.wordCountAlert">You have exceeded the maximum word limit for abstracts in this section's track.</message>
```
Comment (NateWr) [inline locale/en_US/submission.xml]: I think you can re-use the string from above. But either way, it should include the maximum word count in the error message.

Comment (bozana) [reply locale/en_US/submission.xml]: Here I am not able to pass the parameter to the keys/messages in the FormValidator, s. https://github.com/pkp/pkp-lib/blob/master/classes/form/validation/FormValidator.inc.php#L78.
So I can leave it as it is?

Comment (NateWr) [reply locale/en_US/submission.xml]: Is it possible to use FormValidatorLength? https://github.com/pkp/pkp-lib/blob/master/classes/form/validation/FormValidatorLength.inc.php

Comment (NateWr) [reply locale/en_US/submission.xml]: > Your abstract is too long. Please shorten it to below the word limit indicated for this section.

**Pull request merged**